RHIDP-5959 Update dangerouslyAllowSignInWithoutUserInCatalog config (#978)
redhat-developer/red-hat-developers-documentation-rhdh: modules/authentication/proc-enabling-authentication-with-rhbk.adoc

`AUTH_OIDC_METADATA_URL`:: Enter the saved **{rhbk} realm base URL**.

. To set up the {rhbk} authentication provider in your {product-short} custom configuration, edit your custom {product-short} ConfigMap such as `app-config-rhdh`, and add the following lines to the `{my-app-config-file}` content:

.. Configure mandatory fields:
+
.`{my-app-config-file}` fragment with mandatory fields to enable authentication with {rhbk}
[source,yaml]
----
auth:
  environment: production # <1>
  providers:
    oidc:
      production:
        metadataUrl: ${AUTH_OIDC_METADATA_URL} # <2>
        clientId: ${AUTH_OIDC_CLIENT_ID}
        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
signInPage: oidc # <3>
----

> <1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.

Author: ```suggestion
<1> Mark the environment as `production` to hide the Guest login in the {product-short} login page.
```

> <2> Apply the same {rhbk} credentials configured in your {product-short} secrets.

Reviewer: ```suggestion
<2> Apply the {rhbk} credentials configured in your {product-short} secrets.
```

> <3> To enable the OIDC provider as default sign-in provider.

Reviewer 2: Please make this consistent with the other providers :)

> <1> Mark the environment as `production` to hide the Guest login in the {product-short} login page.
> <2> Configure the OIDC provider with your secrets.
> <3> To enable the OIDC provider as default sign-in provider.

Author: ```suggestion
<1>  Mark the environment as `production` and disable the Guest login option in the {product-short} login page.
<2> Apply the same {rhbk} credentials configured in your {product-short} secrets.
<3> To enable the {rhbk} provider as your {product-short} sign-in provider.
```

      production:
        scope: ${AUTH_OIDC_SCOPE}
----

`signIn`::
`resolvers`:::
Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.

> The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed.

Author: After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
+
The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
+
WARNING: In production mode, only configure one resolver to ensure users are securely matched.

Author: ```suggestion
After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
+
The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
+
WARNING: In production mode, only configure one resolver to ensure users are securely matched. 
```

      production:
        scope: ${AUTH_OIDC_SCOPE}
----

`signIn`::
`resolvers`:::

> After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.

Reviewer: ```suggestion
After successfully authenticating, the signing-in user must be matched to an existing user in the {product-short} catalog. To securely match users for your use case, you can configure a specific resolver. Add the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
```

Author: Same as above.

            - resolver: emailMatchingUserEntityProfileEmail
            - resolver: emailLocalPartMatchingUserEntityName
----

`dangerouslyAllowSignInWithoutUserInCatalog: true`::::

> To enable authentication without requiring to provision users in the {product-short} software catalog.

Author: ```suggestion
Configure the sign-in resolver to bypass the user provisioning requirement in the {product-short} software catalog.
```

+
WARNING: Use this option to explore {product-short} features, but do not use it in production.
+

> .`{my-app-config-file}` fragment with optional field to allow authenticating users absent from the software catalog

Author: ```suggestion
.`app-config-rhdh.yaml` fragment with optional field to allow signing in users absent from the software catalog
```
